exporter/datadogexporter/internal/metadata/internal/gohai/doc.go

```diff
@@ -0,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
I don't think this warrants its own package. Why did you decide to do that? Please note that the more packages you add to a module, the more the chances of introducing circular dependencies grows, and fixing them is hard.

New packages should only be introduced if they must be new packages, if they make sense as part of a bigger functionality. This doesn't seem to be the case. I'd argue that this should be part of `internal/metadata`

Reply:
I was following the structure we have in datadog agent . We can move code it to `internal/metadata`  that works too